tests/apollo/util/bft_debug_tool.py

```diff
@@ -0,0 +1,144 @@
+            i = 0
+            while i < len(sanitizer_options):
+                if "log_path=" in sanitizer_options[i]: # remove the older log path
+                    i = i + 2
+                else:
+                    sanitizer_options_env_val += sanitizer_options[i]
+                    i = i + 1
+            sanitizer_options_env_val += ":log_path=" + os.path.join(self.output_log_path, self.output_log_prefix)
+            os.environ[sanitizer_options_env_key] = sanitizer_options_env_val
+        # re-create the test output folder (or create if it was never there). Same as log output, we delete everything from old tests
+        shutil.rmtree(self.output_log_path, ignore_errors=True)
+        os.makedirs(self.output_log_path, exist_ok=True)
+
+    def process_pids_after_replica_started(self, started_proc_info, started_replica_id):
+        '''
+        Some tools may spawn skvbc_replica as child process. We need to keep track of the process information to be able to signal
+        for termination at later stages.
+        Returns: the tester replica process info  (according to the debug tool)
+        '''
+        assert self.name is not None, "This function should be called only if there is an active debug tool!"
+        if self.name == "HEAPTRACK":
+            # proc holds heaptrack subprocess object.
+            # At later stage we would like to signal the actual replica executable to stop, so lets find its pid using psutil
+            for child_proc in psutil.Process(started_proc_info.pid).children():
+                if child_proc.name() == 'skvbc_replica':
+                    ret = psutil.Process(child_proc.pid)
+                    self.external_tool_procs[started_replica_id] = started_proc_info   # external_tool_procs holds Heaptrack's process info
+                    break
+            assert started_replica_id in self.external_tool_procs
+            return ret
+        else:
+            return started_proc_info
+
+    def process_output(self, stopped_replica_id, stoppped_proc, testdir):
+        '''
+        Copy and/or extract and/or rename the debug tool logs into the correct output output folder
+        '''
+        output_log_format = "%s_skvbc_replica_pid_%d_repid_%d_#%d.%s" # tool name, process id, replica id, counter, postfix
+        if stopped_replica_id not in self.output_counters:
+            self.output_counters[stopped_replica_id] = 0    # May happen in reconfiguration
+        counter = self.output_counters[stopped_replica_id]
+        if self.name == "HEAPTRACK":
+            # Heaptrack output is a gz file. We would like to rename the gz file and extract it to a log file with heaptrack_print
+            assert (stopped_replica_id in self.external_tool_procs)
+            heaptrack_proc = self.external_tool_procs[stopped_replica_id]
+            heaptrack_proc.wait()
+            heaptrack_src_log_name = "heaptrack.skvbc_replica." + str(heaptrack_proc.pid) + ".gz"
```

Comment:
why not use python3 format strings (everywhere)?

Reply:
Well yes, this code was written in few iterations and time. Looks wierd, few lines up i'm using format and here not. I'll fix it.
thanks.

Reply:
Thanks, I fixed in 2-3 places in the file.